test/functional/core/path_spec.lua

```diff
@@ -130,3 +130,40 @@ describe('file search (gf, <cfile>)', function()
     test_cfile([[\\127.0.0.1\c$\temp\test-file.txt]], [[127.0.0.1]], [[\\127.0.0.1\c$\temp\test-file.txt]])
   end)
 end)
+
+describe('file search with vim functions', function()
+  local test_folder = "path_spec_folder"
+
+  setup(function()
+    mkdir(test_folder)
+  end)
+
+  teardown(function()
+    rmdir(test_folder)
+  end)
+
+  ---@param option "dir" | "file"
+  local function test_find_func(option, folder, item)
+    local folder_path = join_path(test_folder, folder)
+    mkdir(folder_path)
+    local expected = join_path(folder_path, item)
+    if option == "dir" then
+      mkdir(expected)
+    else
+      write_file(expected, '')
+    end
+    eq(expected, eval('find'..option..'(fnameescape(\''..item..'\'),fnameescape(\''..folder_path..'\'))'))
```

Comment:
Nit: this is unnecessarily abstracted/indirect. The caller can just pass "finddir" or "findfile".